system/smbus.c

```diff
@@ -297,27 +297,28 @@ static spd_info parse_spd_ddr5(uint8_t slot_idx)
         // CAS# Latency
         tns  = (uint16_t)get_spd(slot_idx, 718 + xmp_offset) << 8 |
                (uint16_t)get_spd(slot_idx, 717 + xmp_offset);
-        spdi.tCL = (uint16_t)(tns/tCK + 0.5f);
+        spdi.tCL = (tns + tCK - 1) / tCK;
+        spdi.tCL += (spdi.tCL % 2 != 0) ? 1 : 0; // if tCL is odd, round to upper even.
```

Comment:
For modulo 2, that's equivalent to `spdi.tCL += (spdi.tCL % 2 == 1) ? 1 : 0`, i.e. the ternary is superfluous. It may be clearer to write `spdi.tCL += spdi.tCL % 2`, keeping the comment, of course.

Reply:
Right!